File: backend/src/main/java/com/backend/domain/analysis/service/SseProgressNotifier.java
package com.backend.domain.analysis.service;

import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.stereotype.Component;

@Slf4j
@Component
@RequiredArgsConstructor
public class SseProgressNotifier implements ProgressNotifier {
    private final AnalysisProgressService progressService;

    @Override
    public void notify(Long userId, String event, String message) {
        if(userId != null) {  // 수정!
            log.debug("[SSE] userId={}, event={}, message={}", userId, event, message);
            progressService.sendEvent(userId, event, message);
        } else {
            log.warn("[SSE] userId가 null입니다. 이벤트 전송 실패: {}", event);
        }
    }
}

File: backend/src/main/java/com/backend/domain/analysis/service/SseProgressNotifier.kt
package com.backend.domain.analysis.service

import org.slf4j.LoggerFactory
import org.springframework.stereotype.Component

@Component
class SseProgressNotifier(
    private val progressService: AnalysisProgressService
) : ProgressNotifier {

    private val log = LoggerFactory.getLogger(SseProgressNotifier::class.java)

    override fun notify(userId: Long, event: String, message: String) {
        log.debug("[SSE] userId={}, event={}, message={}", userId, event, message)
        progressService.sendEvent(userId, event, message)
    }
}
